pkg/skaffold/config/options.go

```diff
@@ -40,6 +40,8 @@ type SkaffoldOptions struct {
 	CacheArtifacts    bool
 	ExperimentalGUI   bool
 	EnableRPC         bool
+	NoPrune           bool
+	Prune             bool // this is a logical combination of NoPrune and CacheArtifacts
```

Comment:
I find this is very confusing - what does it mean that it's a logical combination? 
why do we need to store calculations in a "theoretically" read-only struct for user defined opts?

Reply:
maybe we could implement an instance function on it?

Reply:
`opts.Prune()`?

Reply:
by logical combination I mean that `opts.Prune` == `!opts.NoPrune` && `!opts.CacheArtifacts`:

|  NoPrune | CacheArtifacts |  **Prune** |
| --- | --- | --- |
| F (default) | F (default) | T (default) |
| F (default) | T | F |
| T | F (default) | F |
| T | T | F |

it's not really storing a calculation, it's just storing a simplified version of it to be used later on. to me this makes more sense than doing `opts.Prune()`, WDYT?

Reply:
I got the simplification - that part is fine - but I think we should introduce the `IsPrune()` function similar to `Labels()`  instead of storing it in the struct to keep the design consistent i.e. "fields of SkaffoldOptions are user defined flags".

Reply:
updated